instrumentation/cassandra/cassandra-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/cassandra/v4_0/CassandraDatabaseClientTracer.java

```diff
@@ -49,14 +56,67 @@ protected InetSocketAddress peerAddress(CqlSession cqlSession) {
     return null;
   }
 
+  @Override
+  protected Span onConnection(Span span, CqlSession cqlSession) {
+    span = super.onConnection(span, cqlSession);
+    DriverExecutionProfile config = cqlSession.getContext().getConfig().getDefaultProfile();
+    // may be overwritten by statement, but take the default for now
+    int pageSize = config.getInt(DefaultDriverOption.REQUEST_PAGE_SIZE);
+    if (pageSize > 0) {
+      span.setAttribute(SemanticAttributes.DB_CASSANDRA_PAGE_SIZE, pageSize);
+    }
+    // may be overwritten by statement, but take the default for now
+    span.setAttribute(
+        SemanticAttributes.DB_CASSANDRA_CONSISTENCY_LEVEL,
+        config.getString(DefaultDriverOption.REQUEST_CONSISTENCY));
```

Comment:
that is correct

Reply:
it's a little less dire than that, on exception it will still attempt to extract the execution info and invoke onResponse

Reply:
can we avoid capturing the wrong value? In general it's better to capture no value than to capture the wrong value

Reply:
it seems that actually did not come over on the rebase, I'll add that back in

Reply:
> it's a little less dire than that, on exception it will still attempt to extract the execution info and invoke onResponse

can you point me where you're seeing that? this is what I'm looking at in `TracingCqlSession`, and on exception it doesn't call `onResponse`:

```
      try {
        ResultSet resultSet = session.execute(query);
        tracer().onResponse(context, resultSet.getExecutionInfo());
        return resultSet;
      } catch (RuntimeException e) {
        tracer().endExceptionally(context, e);
        throw e;
      } finally {
        tracer().end(context);
      }
```

Reply:
> it seems that actually did not come over on the rebase, I'll add that back in

👍 (ignore my last comment)

Reply:
what is the best way to clear the attribute on a span @trask? The setAttribute method is annotated with not null.

Reply:
can we set the attributes only once we have the statement?

Reply:
the default values comes out of the session which is only available in onConnection, not in onResponse

Reply:
can we pass the session in to `onResponse`?

Reply:
we can, but it means the endExceptionally method no longer overrides the superclass method since it also needs the session. I have added that an implementation that throws an illegal state exception if called.

another option would be to pass the session and execution info as a pair, but that may create undesired garbage.

Reply:
hey @trask, are you happy with the change to add the session into onResponse meaning it also needs to be passed into endExceptionally?